gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -170,7 +170,7 @@ class {{service.name}}RestTransport({{service.name}}Transport):
     {% if method.input.required_fields %}
     __{{ method.name | snake_case }}_required_fields_default_values =  {
     {% for req_field in method.input.required_fields if req_field.is_primitive %}
-        "{{ req_field.name | camel_case }}" : {% if req_field.field_pb.default_value is string %}"{{req_field.field_pb.default_value }}"{% else %}{{ req_field.field_pb.default_value }}{% endif %},{# default is str #}
+        "{{ req_field.name | camel_case }}" : {% if req_field.field_pb.type == 9 %}"{{req_field.field_pb.default_value }}"{% else %}{{ req_field.type.python_type(req_field.field_pb.default_value or 0) }}{% endif %},{# default is str #}
```

Comment:
`req_field.field_pb.type == 9` feels strange to me. I assume it is from https://googleapis.dev/python/protobuf/latest/google/protobuf/descriptor_pb2.html#google.protobuf.descriptor_pb2.FieldDescriptorProto.TYPE_STRING. Can't we use that constant directly? is there any other way to do this?

Can't this python_type function do the work of passing returning the correct thing depending on what type we are working with?

Reply:
It's difficult and messy to do module import and access from the template logic. I agree that using magic numbers here is suboptimal, but it's the better solution to a nasty problem.

`python_type` returns a Python type, e.g. `float`, `int`, `bool`. What I'm using it here for is as a constructor and giving it a _real_ default of 0, which turns into `0`, `0.0`, or `False`.